### ISSUE TITLE
Global search include filter does not work with folder names

### ISSUE THREAD

**Issue by RSeidelsohn** [zed-industries/zed#21804]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

This is an issue on my Linux version of Zed 0.164.2 since a few recent versions:

Using the global search within a project folder structure with a file ending filter works fine, but when I try to restrict the search using the "include" filter with a subfolder name, I get no results any more. This used to work fine before:

The "include" filter `*.hbs` gives results only for handlebars files, but the "include" filter of say, `subfolder-name/**/*.hbs` stays empty, even though _without_ the directory name, it gives me also results below the folder `subfolder-name`.

For the log attached, I started Zed and did a search with the folder name in the "include" filter, giving me no result where there should be many results.

### Environment

Zed: v0.164.2 (Zed)
OS: Linux X11 ubuntu 24.04
Memory: 30.6 GiB
Architecture: x86_64
GPU: AMD Radeon Graphics (RADV REMBRANDT) || radv || Mesa 24.0.9-0ubuntu0.2

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary>

```
2024-12-10T14:13:08.411836686+01:00 [INFO] ========== starting zed ==========
2024-12-10T14:13:08.443015548+01:00 [INFO] Compositor detection: _NET_WM_CM_S?=false, _NET_WM_CM_OWNER=false, _NET_SUPPORTING_WM_CHECK=true
2024-12-10T14:13:08.443098505+01:00 [INFO] x11: compositor present: true, gtk_frame_extents_supported: true
2024-12-10T14:13:08.450728567+01:00 [INFO] Try connect ibus
2024-12-10T14:13:08.451577734+01:00 [INFO] perform;
2024-12-10T14:13:08.451632889+01:00 [INFO] authenticate;
2024-12-10T14:13:08.451700006+01:00 [INFO] read_command;
2024-12-10T14:13:08.451747155+01:00 [INFO] Opening main db
2024-12-10T14:13:08.452493177+01:00 [INFO] new;
2024-12-10T14:13:08.452543993+01:00 [INFO] socket reader;
2024-12-10T14:13:08.453013331+01:00 [INFO] Opening main db
2024-12-10T14:13:08.453537003+01:00 [INFO] keep_updated;
2024-12-10T14:13:08.454992658+01:00 [INFO] Using git binary path: None
2024-12-10T14:13:08.49735621+01:00 [INFO] No prompt template overrides directory found at /home/xyzabc/.local/share/zed/prompt_overrides. Using built-in prompts.
2024-12-10T14:13:08.498227809+01:00 [INFO] extensions updated. loading 15, reloading 0, unloading 0
2024-12-10T14:13:08.500518074+01:00 [INFO] activate is not implemented on Linux, ignoring the call
2024-12-10T14:13:08.502764836+01:00 [INFO] Opening main db
2024-12-10T14:13:08.503167689+01:00 [INFO] perform;
2024-12-10T14:13:08.503216201+01:00 [INFO] authenticate;
2024-12-10T14:13:08.503277827+01:00 [INFO] read_command;
2024-12-10T14:13:08.504095996+01:00 [INFO] socket reader;
2024-12-10T14:13:08.511013188+01:00 [INFO] new;
2024-12-10T14:13:08.511853419+01:00 [INFO] keep_updated;
2024-12-10T14:13:08.518800537+01:00 [INFO] set status on client 0: Authenticating
2024-12-10T14:13:08.51926154+01:00 [INFO] perform;
2024-12-10T14:13:08.519306776+01:00 [INFO] authenticate;
2024-12-10T14:13:08.519357622+01:00 [INFO] read_command;
2024-12-10T14:13:08.520311758+01:00 [INFO] socket reader;
2024-12-10T14:13:08.521267727+01:00 [INFO] Using Visual { id: 276, colormap: 0, depth: 32 }
2024-12-10T14:13:08.521330255+01:00 [INFO] Creating colormap 69206019
2024-12-10T14:13:08.523665415+01:00 [WARN] Theme "Colorizer" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.523713266+01:00 [WARN] Theme "Colorizer Pro" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.524304966+01:00 [INFO] building git repository, `.git` path in the worktree: ".git"
2024-12-10T14:13:08.527603258+01:00 [INFO] new;
2024-12-10T14:13:08.52860775+01:00 [INFO] keep_updated;
2024-12-10T14:13:08.549303701+01:00 [INFO] Enabling Vulkan Portability
2024-12-10T14:13:08.5493612+01:00 [INFO] Enabling color space support
2024-12-10T14:13:08.574563881+01:00 [INFO] Adapter: "AMD Radeon Graphics (RADV REMBRANDT)"
2024-12-10T14:13:08.575769042+01:00 [INFO] Ray tracing is supported
2024-12-10T14:13:08.585852477+01:00 [INFO] Using surface present mode MAILBOX
2024-12-10T14:13:08.585971232+01:00 [WARN] Unable to forbid exclusive full screen
2024-12-10T14:13:08.59019681+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-12-10T14:13:08.590299634+01:00 [INFO] Creating a descriptor pool for at most 16 sets
2024-12-10T14:13:08.590357364+01:00 [INFO] Initializing Blade pipelines for surface SurfaceInfo { format: Bgra8UnormSrgb, alpha: Ignored }
2024-12-10T14:13:08.63083165+01:00 [INFO] Refreshing at 16663 micros
2024-12-10T14:13:08.632028044+01:00 [INFO] Using GPU: GPUSpecs { is_software_emulated: false, device_name: "AMD Radeon Graphics (RADV REMBRANDT)", driver_name: "radv", driver_info: "Mesa 24.0.9-0ubuntu0.2" }
2024-12-10T14:13:08.639071485+01:00 [INFO] XConnected server on 29360137, transport version: 0.0, TRANSPORT_MAX: 20
2024-12-10T14:13:08.639334453+01:00 [INFO] set status on client 200125: Connecting
2024-12-10T14:13:08.645797416+01:00 [INFO] Opening main db
2024-12-10T14:13:08.656074497+01:00 [INFO] Opening main db
2024-12-10T14:13:08.663753882+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Dark I" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.663823955+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Dark II" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.663859832+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Dark P1" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.663894538+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Dark P31" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.663934824+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Dark P39 I" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.663968819+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Dark P39 II" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.664002543+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Light" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.664036317+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Light P1" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.66406978+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Light P31" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.664103243+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Light P39 I" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.664136426+01:00 [WARN] Theme "Green Monochrome Monitor CRT Phosphor Light P39 II" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.680177056+01:00 [WARN] Theme "Material Dark" is using a deprecated style property: scrollbar_thumb.background. Use `scrollbar.thumb.background` instead.
2024-12-10T14:13:08.705366002+01:00 [ERROR] missing field `name` at line 934 column 1
2024-12-10T14:13:08.783825282+01:00 [INFO] attempting to start language server "tailwindcss-language-server", path: "/home/xyzabc/Development/test.html", id: 1
2024-12-10T14:13:08.783908941+01:00 [INFO] Initializing default prettier with plugins {}
2024-12-10T14:13:08.790513531+01:00 [INFO] Initializing default prettier with plugins {}
2024-12-10T14:13:08.803661607+01:00 [INFO] connected to rpc endpoint https://collab.zed.dev/rpc
2024-12-10T14:13:08.812025988+01:00 [INFO] fetching latest version of language server "tailwindcss-language-server"
2024-12-10T14:13:08.817363049+01:00 [INFO] Initializing default prettier with plugins {}
2024-12-10T14:13:08.819308001+01:00 [INFO] perform;
2024-12-10T14:13:08.819369277+01:00 [INFO] authenticate;
2024-12-10T14:13:08.819433939+01:00 [INFO] read_command;
2024-12-10T14:13:08.820488265+01:00 [INFO] socket reader;
2024-12-10T14:13:08.836877224+01:00 [INFO] Initializing default prettier with plugins {}
2024-12-10T14:13:09.196101602+01:00 [INFO] add_connection;
2024-12-10T14:13:09.21514765+01:00 [INFO] set status on client 200125: Connected { peer_id: PeerId { owner_id: 622, id: 540898 }, connection_id: ConnectionId { owner_id: 0, id: 0 } }
2024-12-10T14:13:09.263820224+01:00 [INFO] set environment variables from shell:/usr/bin/zsh, path:/home/xyzabc/.bun/bin:/home/xyzabc/.asdf/shims:/home/xyzabc/.asdf/bin:/home/xyzabc/.cargo/bin:/home/xyzabc/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/xyzabc/.rvm/bin:/home/xyzabc/.local/share/JetBrains/Toolbox/scripts:/home/xyzabc/.rvm/bin
2024-12-10T14:13:09.395997587+01:00 [INFO] perform;
2024-12-10T14:13:09.396055036+01:00 [INFO] authenticate;
2024-12-10T14:13:09.396091224+01:00 [INFO] read_command;
2024-12-10T14:13:09.396776091+01:00 [INFO] socket reader;
2024-12-10T14:13:09.472823496+01:00 [INFO] using project environment variables shell launched in "/home/xyzabc/Development/test.html". PATH="/home/xyzabc/.bun/bin:/home/xyzabc/.asdf/shims:/home/xyzabc/.asdf/bin:/home/xyzabc/.cargo/bin:/home/xyzabc/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin:/home/xyzabc/.rvm/bin:/home/xyzabc/.local/share/JetBrains/Toolbox/scripts:/home/xyzabc/.rvm/bin"
2024-12-10T14:13:09.531849806+01:00 [INFO] Initializing default prettier with plugins {}
2024-12-10T14:13:09.53363262+01:00 [INFO] perform;
2024-12-10T14:13:09.533686272+01:00 [INFO] authenticate;
2024-12-10T14:13:09.533739443+01:00 [INFO] read_command;
2024-12-10T14:13:09.546637695+01:00 [INFO] socket reader;
2024-12-10T14:13:10.21883964+01:00 [INFO] starting language server process. binary path: "/home/xyzabc/.asdf/shims/node", working directory: "/", args: ["/home/xyzabc/.local/share/zed/copilot/copilot-v0.7.0/dist/language-server.js", "--stdio"]
2024-12-10T14:13:10.42585996+01:00 [INFO] downloading language server "tailwindcss-language-server"
2024-12-10T14:13:10.498354401+01:00 [INFO] Initializing default prettier with plugins {}
2024-12-10T14:13:10.498420927+01:00 [INFO] attempting to start language server "emmet-language-server", path: "/home/xyzabc/Development/test.html", id: 2
2024-12-10T14:13:10.49846485+01:00 [INFO] attempting to start language server "vscode-html-language-server", path: "/home/xyzabc/Development/test.html", id: 3
2024-12-10T14:13:10.498496169+01:00 [INFO] attempting to start language server "snippet-completion-server", path: "/home/xyzabc/Development/test.html", id: 4
2024-12-10T14:13:10.499990688+01:00 [INFO] perform;
2024-12-10T14:13:10.500036585+01:00 [INFO] authenticate;
2024-12-10T14:13:10.500074086+01:00 [INFO] read_command;
2024-12-10T14:13:10.5007526+01:00 [INFO] socket reader;
2024-12-10T14:13:10.772023699+01:00 [INFO] starting language server process. binary path: "/home/xyzabc/.local/share/zed/extensions/work/snippets/simple-completion-language-server-v0.1.1/simple-completion-language-server", working directory: "/home/xyzabc/Development", args: []
2024-12-10T14:13:11.072670407+01:00 [INFO] Language server with id 0 sent unhandled notification window/logMessage:
{
  "type": 3,
  "message": "[lsp] GitHub Copilot Language Server 1.236.0 initialized"
}
2024-12-10T14:13:11.072785545+01:00 [INFO] Language server with id 0 sent unhandled notification workspace/configuration:
{
  "items": [
    {
      "section": "github.copilot"
    },
    {
      "section": "github-enterprise"
    },
    {
      "section": "http"
    },
    {
      "section": "telemetry"
    }
  ]
}
2024-12-10T14:13:11.662815804+01:00 [INFO] starting language server process. binary path: "/home/xyzabc/.asdf/shims/node", working directory: "/home/xyzabc/Development", args: ["/home/xyzabc/.local/share/zed/languages/tailwindcss-language-server/node_modules/.bin/tailwindcss-language-server", "--stdio"]
2024-12-10T14:13:12.888449251+01:00 [INFO] starting language server process. binary path: "/home/xyzabc/.asdf/shims/node", working directory: "/home/xyzabc/Development", args: ["/home/xyzabc/.local/share/zed/extensions/work/emmet/node_modules/.bin/emmet-language-server", "--stdio"]
2024-12-10T14:13:13.632750895+01:00 [WARN] unhandled capability registration: Registration { id: "168b9136-3db4-4069-bdc2-4e051406b739", method: "workspace/didChangeWorkspaceFolders", register_options: Some(Object {}) }
2024-12-10T14:13:13.741326047+01:00 [INFO] starting language server process. binary path: "/home/xyzabc/.asdf/shims/node", working directory: "/home/xyzabc/Development", args: ["/home/xyzabc/.local/share/zed/extensions/work/html/node_modules/@zed-industries/vscode-langservers-extracted/bin/vscode-html-language-server", "--stdio"]
2024-12-10T14:13:14.148667651+01:00 [WARN] unhandled capability registration: Registration { id: "31fefbb5-0890-44a4-b3f4-2cba2b263a56", method: "workspace/didChangeWorkspaceFolders", register_options: Some(Object {}) }

```
</details>

**Comment by 0xtimsb** [comment]
I'm unable to reproduce the issue. There might be a misunderstanding on my part. Do you mean something like this? Could you share a screenshot?

![image](https://github.com/user-attachments/assets/eaa3a641-a457-45a6-9be8-97d51d23db3e)

**Comment by RSeidelsohn** [comment]
Exactly this - though for me it won't show any search results (any more) - I get those results when I either filter for `*.astro` or `src` alone. 
But I figured out one more thing: If I filter for `src`, it searches below the `src` folder only, which is fine. But if I add a trailing slash like so: `src/` or even `src/*`, I don't get any result. Same would be for (imaginary folder structure): `src/images` would work fine, but `src/images/` not. It's as if I would not be allowed to append any slash after the last folder name.
I'll prepare a screenshot once I'm on my Linux system.

**Comment by RSeidelsohn** [comment]
It's solved now. It was specific to one of my projects and it turned out that the reason it was malfunctioning whenever I added a directory separator (`/`) into the _include_  filter, was the fact that I had an HTML file in the project that I moved into the same folder as the project root folder, so that in the project panel I saw the project root folder _and_ the single HTML file that resided in the same folder as the project root folder was residing in.

This somehow broke the search. After I removed that single HTML file from the project with the "remove from project" command, the search was working again just as it should.

So this was a strange edge case I would not really consider a bug or at least not an important one.

Sorry for all the trouble, but possibly an interesting finding.

Cheers, Roman.
